components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -100,6 +109,35 @@ ESP8266Interface::~ESP8266Interface()
     }
 }
 
+ESP8266Interface::ResetPin::ResetPin(mbed::DigitalOut rst_pin) : _rst_pin(rst_pin)
+{
+}
+
+void ESP8266Interface::ResetPin::assert()
+{
+    if (_rst_pin.is_connected()) {
+        _rst_pin.write(0);
+        // If you happen to use Pin7 CH_EN as reset pin, not needed otherwise
+        // https://www.espressif.com/sites/default/files/documentation/esp8266_hardware_design_guidelines_en.pdf
+        wait_us(200);
```

Comment:
[just a thought]
Is there a way to check which pin was used? Some ifdef against a compilation flag or perhaps we could look inside the _rst_pin and determine it?
Then we could wrap the delay in an if clause and only call it when Pin7 is used.

Reply:
Probably could but I would say it's not worth the hassle.

Reply:
Not sure why you want a delay on the assert. If doing a reset pulse, you'd want a delay in between the assert and deassert, but if just asserting to "shut it down", you'd not want the delay.

If anything I'd have thought you'd want a delay after deassert.

Reply:
After discussing with Kevin the desicion was to leave this as it is.